test/integration/update_int_test.go

```diff
@@ -143,6 +144,39 @@ func (suite *UpdateIntegrationTestSuite) testUpdate(ts *e2e.Session, baseDir str
 	cp.Expect("Installing Update")
 }
 
+func (suite *UpdateIntegrationTestSuite) TestUpdate_Repair() {
+	suite.OnlyRunForTags(tagsuite.Update)
+	ts := e2e.New(suite.T(), true)
+	defer ts.Close()
+
+	ts.UseDistinctStateExes()
+
+	cfg, err := config.NewCustom(ts.Dirs.Config, singlethread.New(), true)
+	suite.Require().NoError(err)
+	defer cfg.Close()
+
+	subBinDir := filepath.Join(ts.Dirs.Bin, "bin")
+	ts.CopyExeToDir(ts.Exe, filepath.Join(subBinDir, filepath.Base(ts.Exe)))
+	ts.CopyExeToDir(ts.SvcExe, filepath.Join(subBinDir, filepath.Base(ts.SvcExe)))
+	ts.CopyExeToDir(ts.TrayExe, filepath.Join(subBinDir, filepath.Base(ts.TrayExe)))
```

Comment:
Let's copy recursively from the dir that `ts.Exe` is in, that might help surface future edge-cases.

Reply:
Could not use `fileutils.CopyFiles` as it lead to recursion issues. Instead, I'm just reading the files in the `bin` directory and copying those.